data_model/src/query.rs

```diff
@@ -1522,9 +1530,68 @@ pub mod block {
         }
     }
 
+    #[derive(
+        Default,
+        Debug,
+        Clone,
+        Copy,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    /// `FindAllBlockHeaders` Iroha Query will list all block headers
+    pub struct FindAllBlockHeaders;
+
+    impl Query for FindAllBlockHeaders {
+        type Output = Vec<BlockHeaderValue>;
+    }
+
+    impl FindAllBlockHeaders {
+        /// Construct [`FindAllBlockHeaders`].
+        pub fn new() -> Self {
+            Self {}
+        }
+    }
+
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    /// `FindBlockHeaderByHash` Iroha Query will find block header by block hash
+    pub struct FindBlockHeaderByHash {
+        /// Block hash hash.
```

Comment:
```suggestion
        /// The hash of the block hash.
```

Reply:
It's a typo actually, it's should be just one layer of hash. Thanks for catching.